march_joint_inertia_controller/src/joint_trajectory_inertia_controller.cpp

```diff
@@ -0,0 +1,18 @@
+// Copyright 2020 Project March.
+#include <angles/angles.h>
+#include <control_msgs/FollowJointTrajectoryAction.h>
+#include "march_joint_inertia_controller/joint_trajectory_inertia_controller.h"
+#include <math.h>
+#include <pluginlib/class_list_macros.h>
+#include <pluginlib/class_list_macros.hpp>
+#include <trajectory_msgs/JointTrajectory.h>
+
+namespace inertia_trajectory_controller
+{
+typedef joint_trajectory_controller::JointTrajectoryController<trajectory_interface::QuinticSplineSegment<double>,
+                                                               hardware_interface::EffortJointInterface>
+    JointTrajectoryController;
+
+}  // namespace inertia_trajectory_controller
+
+PLUGINLIB_EXPORT_CLASS(inertia_trajectory_controller::JointTrajectoryController, controller_interface::ControllerBase);
```

Comment:
So the `intertia_trajectory_controller` is just exactly the same as the effort controller? Since you export it as the `EffortJointInterface?

Reply:
not exactly, the main difference lies in that the trajectory controller gets fed its reference values by the FollowJointTrajectory actions. these are then given through to the controller. The effort controller can have any self defined interface i.e. not reliant on a gait message.